src/gps/dateTime.c

```diff
@@ -133,3 +133,118 @@ tiny_millis_t getUptime() {
 int getUptimeAsInt() {
    return (int) getUptime();
 }
+
+void getDateTimeFromEpochMillis(DateTime *dateTime, millis_t millis) {
```

Comment:
I have to agree with Brent on this one.  Passing in struct pointers to a function for use as a return is an INCREDIBLY common C programming paradigm.  It's everywhere.  This isn't complicated or tricky, and arguably if other people looking at this code have no idea that this is a thing, they probably aren't the people you want contributing to your codebase.  

To your memory isolationist comment: Upping stack usage for us has the exact opposite effect of reliability.  Dealing with stack overflow problems is incredibly frustrating on embedded platforms, and many of them will come up and bite you in the ass without ever knowing it.  This is in no way 'more reliable'.

When do we start getting anal retentive with stack space?  Always.  I think you're overstating the cost/benefit here.

This hardly feels like premature optimization, it is a coding style preference.  When I saw your original comment, my first thought was "Who the fuck passes a structure by value? That's terrible for stack useage".  What if this function could fail?  How would we return an error condition?  We'd be in the same boat as we are now, but what would your argument be?